src/libutil/util.cc

```diff
@@ -1718,6 +1747,20 @@ void startSignalHandlerThread()
 
 static void restoreSignals()
 {
+    // If startSignalHandlerThread wasn't called, that means we're not running
+    // in a proper libmain process, but a process that presumably manages its
+    // own signal handlers. Such a process should call either
+    //  - initNix(), to be a proper libmain process
+    //  - startSignalHandlerThread(), to resemble libmain regarding signal
+    //    handling only
+    //  - saveSignalMask(), for processes that define their own signal handling
+    //    thread
+    // TODO: Warn about this? Have a default signal mask? The latter depends on
+    //       whether we should generally inherit signal masks from the caller.
+    //       I don't know what the larger unix ecosystem expects from us here.
+    if (!savedSignalMaskIsSet)
+        return;
```

Comment:
@edolstra, would it make sense to define a default signal mask for child processes, or is a unix process like `nix` expected to inherit the signal mask of its caller _and_ pass that mask on to its children (such as `git`, `tar`, ...)?

If you think it's sensible to unblock all signals in child processes, I would do this to make the library a bit more robust / work better out of the box.

```suggestion
    if (!savedSignalMaskIsSet)
        sigemptyset(&savedSignalMaskIsSet);
```

Reply:
If you don't know whether it's acceptable unixing to unblock all signals in child processes, that's ok too; we'd just stick to the current, conservative approach.